File: todo-reservation-springboot/src/main/java/de/vkb/dojo/es/facilityManagement/model/event/RoomUpdated.java
package de.vkb.dojo.es.facilityManagement.model.event;

import com.fasterxml.jackson.annotation.*;
import de.vkb.dojo.es.common.model.Change;

@JsonIgnoreProperties(ignoreUnknown = true)
@JsonTypeName("room-updated")
public class RoomUpdated extends RoomEvent {
    private final String operationId;
    private final String aggregateId;

    private final Change<String> name;

    @JsonCreator
    public RoomUpdated(
            @JsonProperty("operatonId") String operationId,
            @JsonProperty("aggregateId") String aggregateId,
            @JsonProperty("name") Change<String> name
    ) {
        this.operationId = operationId;
        this.aggregateId = aggregateId;
        this.name = name;
    }

    @Override
    @JsonGetter
    public String getOperationId() {
        return operationId;
    }

    @Override
    @JsonGetter
    public String getAggregateId() {
        return aggregateId;
    }

    @JsonGetter
    public Change<String> getName() {
        return name;
    }
}

File: todo-reservation-micronaut/src/main/kotlin/de/vkb/dojo/es/facilityManagement/model/event/RoomUpdated.kt
package de.vkb.dojo.es.facilityManagement.model.event

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonTypeName
import de.vkb.dojo.es.common.model.Change

@JsonIgnoreProperties(ignoreUnknown = true)
@JsonTypeName("room-updated")
data class RoomUpdated(
    override val operationId: String,
    override val aggregateId: String,
    val name: Change<String>?
): RoomEvent
